deployment/ccip/changeset/v1_6/cs_ccip_home.go

```diff
@@ -376,63 +380,50 @@ func PromoteCandidateChangeset(
 	if err != nil {
 		return cldf.ChangesetOutput{}, fmt.Errorf("fetch node info: %w", err)
 	}
+	nonBootstraps := nodes.NonBootstraps()
 
-	txOpts := e.BlockChains.EVMChains()[cfg.HomeChainSelector].DeployerKey
-	if cfg.MCMS != nil {
-		txOpts = cldf.SimTransactOpts()
-	}
-
-	homeChain := e.BlockChains.EVMChains()[cfg.HomeChainSelector]
-
-	var mcmsTxs []mcmstypes.Transaction
+	dons := make([]ccipseqs.DONUpdatePromotion, 0)
 	for _, plugin := range cfg.PluginInfo {
 		for _, donID := range donIDs {
-			promoteCandidateOps, err := promoteCandidateForChainOps(
-				e.Logger,
-				txOpts,
-				homeChain,
-				state.Chains[cfg.HomeChainSelector].CapabilityRegistry,
-				state.Chains[cfg.HomeChainSelector].CCIPHome,
-				nodes.NonBootstraps(),
-				donID,
-				plugin.PluginType,
-				plugin.AllowEmptyConfigPromote,
-				cfg.MCMS != nil,
-			)
+			digest, err := state.Chains[cfg.HomeChainSelector].CCIPHome.GetCandidateDigest(nil, donID, uint8(plugin.PluginType))
 			if err != nil {
-				return cldf.ChangesetOutput{}, fmt.Errorf("generating promote candidate mcms txs: %w", err)
+				return cldf.ChangesetOutput{}, err
+			}
+			if digest == [32]byte{} && !plugin.AllowEmptyConfigPromote {
```

Comment:
Interesting would like to understand how this could switch up the order of operations